persistence/persistence-jdbc/src/test/java/uk/gov/justice/services/jdbc/persistence/PreparedStatementWrapperTest.java

```diff
@@ -0,0 +1,180 @@
+package uk.gov.justice.services.jdbc.persistence;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PreparedStatementWrapperTest {
+
+    @Mock
+    private Connection connection;
+
+    @Mock
+    private PreparedStatement preparedStatement;
+
+    @Mock
+    private ResultSet resultSet;
+
+    @Test
+    public void shouldCloseConnectionWhenPreparedStatementCreationCausesException() throws SQLException {
+        final String query = "someQuery";
+        when(connection.prepareStatement(query)).thenThrow(new SQLException());
+        try {
+            PreparedStatementWrapper.valueOf(connection, query);
+        } catch (Exception e) {
+
+        }
+
+        verify(connection).close();
+    }
+
+    @Test
+    public void shouldCloseConnectionAndStatementWhenQueryExecutionThrowsException() throws SQLException {
+        final String query = "someQuery2";
+        when(connection.prepareStatement(query)).thenReturn(preparedStatement);
+        when(preparedStatement.executeQuery()).thenThrow(new SQLException());
+        final PreparedStatementWrapper ps = PreparedStatementWrapper.valueOf(connection, query);
+        try {
+            ps.executeQuery();
+        } catch (Exception e) {
+
+        }
+
+        verify(connection).close();
+        verify(preparedStatement).close();
+    }
+
+    @Test
+    public void shouldCloseConnectionStatementAndResultset() throws SQLException {
+        final String query = "someQuery3";
+        when(connection.prepareStatement(query)).thenReturn(preparedStatement);
+        when(preparedStatement.executeQuery()).thenReturn(resultSet);
+
+
+        final PreparedStatementWrapper ps = PreparedStatementWrapper.valueOf(connection, query);
+        ps.executeQuery();
+
+        ps.close();
```

Comment:
sorry, yea i saw it as a verify not the action. Maybe update the test names to the action/intent not just the resulting expected verify outcomes.